src/scrapper/spiders/slot_spider.py

```diff
@@ -149,35 +154,69 @@ def extractSchedule(self, response):
             end_time = datetime.strptime(schedule["end"], date_format)
 
             for teacher in schedule["persons"]:
+                (sigarra_id, name) = self.get_professor_info(teacher)
+
+                if sigarra_id in self.inserted_teacher_ids:
+                    continue
+
+                self.inserted_teacher_ids.add(sigarra_id)
+
                 yield Professor(
-                    id = teacher["sigarra_id"],
-                    professor_acronym = teacher["acronym"],
-                    professor_name = teacher["name"] #.split("-")[1].strip()
+                    id=sigarra_id,
+                    professor_acronym=teacher["acronym"],
+                    professor_name=name
                 )
 
             for current_class in schedule["classes"]:
                 yield Class(
-                    id=current_class["sigarra_id"],
                     name=current_class["name"],
                     course_unit_id=course_unit_id,
                     last_updated=datetime.now()
                 )
 
+                # INFO Since we need to know at runtime the id of the slot so that we can then create SlotProfessor
+                # instances, we are going to be using the same id as the class in order to minimize database lookups
+                # during the runtime of the scrapper
+                current_class_id = get_class_id(
+                    course_unit_id, current_class["name"])
+
                 yield Slot(
-                    id=current_class["sigarra_id"],
+                    id=current_class_id,
                     lesson_type=schedule["typology"]["acronym"],
                     day=self.days[schedule["week_days"][0]],
                     start_time=start_time.hour + (start_time.minute / 60),
                     duration=(end_time - start_time).total_seconds() / 3600,
                     location=schedule["rooms"][0]["name"],
                     is_composed=len(schedule["classes"]) > 0,
                     professor_id=schedule["persons"][0]["sigarra_id"],
-                    class_id=current_class["sigarra_id"],
+                    class_id=current_class_id,
                     last_updated=datetime.now(),
                 )
-            
+
                 for teacher in schedule["persons"]:
+                    (sigarra_id, name) = self.get_professor_info(
+                        teacher)  # (sigarra_id | None, teacher_name)
+
                     yield SlotProfessor(
-                        slot_id=current_class["sigarra_id"]
-                        professor_id=teacher["sigarra_id"]
+                        slot_id=current_class_id,
+                        professor_id=sigarra_id
```

Comment:
Is `None` a possible value? What happens in these cases?

Reply:
`None` was a possible value in the first version of the `get_professor_info` method at the time that comment was written. 

In the current version, `None` is not a possible value. When the `teacher["name"]` does not respect the format `<professor_id> - <professor_name>`, instead of retrieving the `professor_id` on the left side of the `-`, it justs uses the `teacher["sigarra_id"]` number from the values returned from sigarra api.

To clarify, the `teacher["sigarra_id"]` field is not really the id that appears on the link of the page of the teacher, which is extremely weird but not up to us since we didn't developed that API

Reply:
But the professor_id is used in the link, right?

Reply:
Ok, I just see that there is also a `sigarra_url`, which redirect to teacher profile. Can we save that? I think it could be usefull

Reply:
Yes, the `professor_id` is used in the link.

We can save that, good point!

Reply:
Maybe the URL is safest

Reply:
I'm not entirely sure I understood what you said

Do you mean retrieving the `professor_id` from the `sigarra_url`?

Reply:
No, save the `sigarra_url` in the database as, for example, `teacher_link`

Reply:
I think that maybe it is probably best to do it in a separate issue from this PR, as it is a lower priority issue and this PR should be merged as soon as possible.